### PR TITLE
[Fixes #35]  Add forgot password functionality

### LINKED ISSUE #35: Add forgot password backend route

### PR DESCRIPTION
**Pull request by mahekn23**
# Pull Request

## What does this PR do?
Fixes #35 

Three functions added in authv1/views.py-->
forgot_password - This function sends an email to the user with an otp.
verify_otp - This function verifies the received otp.
update_password - This function accepts a new password and updates it in the database.

## What part does this affect?
- [ ] FrontEnd.
- [x] BackEnd.
- [ ] Documentation.
- [ ] Other. (Please specify below)

## Before submitting
- [x] Was this discussed/approved via a GitHub issue or slack?
- [x] Did you read the [contributor guideline](https://github.com/Auto-DL/Auto-DL/blob/v1-beta/CONTRIBUTING.md)?
- [x] Did you ensure that there aren't any other open [Pull Requests](https://www.github.com/Auto-DL/Generator/pulls) for the same update/change?
- [x] Did you make sure the title is self-explanatory and the description concisely explains the PR?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure the code is clean and docstrings have been added or updated as required?
- [x] Did you make sure the code is linted/formatted locally prior to submission? (using `black` and/or `prettier`)
- [ ] Did you make sure to update the documentation with your changes? (if necessary)

Signed-off-by: mahekn23 2018.mahek.nagdev@ves.ac.in

### CODE REVIEW

File: BackEndApp/authv1/urls.py [hunk 1]
```diff
@@ -6,4 +6,7 @@
     path("login/", views.login, name="login"),
     path("register/", views.register, name="register"),
     path("logout/", views.logout, name="logout"),
+    path("password/forgot/", views.forgot_password, name="forgot_password"),
+    path("password/enter/otp/", views.verify_otp, name="verify_otp"),
```
Comment (shintan777) [inline BackEndApp/authv1/urls.py:10]: do we really need two '/' for /enter/otp? @ADI10HERO

File: BackEndApp/authv1/views.py [hunk 2]
```diff
@@ -93,3 +96,96 @@ def logout(request):
...
+
+@api_view(["POST"])
+def forgot_password(request):
+    try:
+        username = request.data.get("username")
+        user = User(username=username, password=None)
+        this_user = user.find()
+
+        if user is None:
+            message = "User does not exist"
```
Comment (shintan777) [inline BackEndApp/authv1/views.py:109]: Not a good message in terms of security .. Can be something like "Invalid Credentials". @ADI10HERO

File: BackEndApp/authv1/views.py [hunk 3]
```diff
@@ -93,3 +96,96 @@ def logout(request):
...
+def verify_otp(request):
+    try:
+        username = request.data.get("username")
+        received_otp = request.data.get("received_otp")
+        user = User(username=username, password=None)
+        user = user.find()
+
+        otp = OTP(user)
+        otp_verified = otp.verify(received_otp)
+
```
Comment (shintan777) [inline BackEndApp/authv1/views.py:140]: Need to check if the email is verified, before giving a success message

File: BackEndApp/authv1/views.py [hunk 4]
```diff
@@ -93,3 +96,96 @@ def logout(request):
...
+@api_view(["POST"])
+def update_password(request):
+    try:
+        username = request.data.get("username")
+        user = User(username=username, password=None)
+        this_user = user.find()
+
+        if this_user is None:
+            message = "User does not exist"
```
Comment (shintan777) [inline BackEndApp/authv1/views.py:171]: Error message

**Review by shintan777 — CHANGES_REQUESTED** [review]

File: BackEndApp/authv1/views.py (hunk 2 above)
Comment (ADI10HERO) [reply BackEndApp/authv1/views.py:109]: Agreeed 💯

File: BackEndApp/authv1/urls.py (hunk 1 above)
Comment (ADI10HERO) [reply BackEndApp/authv1/urls.py:10]: Having 2 `/` is not a problem tbh @shintan777  but here the names are misleading. 😞 


@mahekn23, as I repeatedly say, naming is one of the hardest things in software engineering and it's absolutely fine to make mistakes here, having something like. But following consistency with the existing codebase + name the routes/fns what they're doing is important.

Comment (mahekn23) [reply BackEndApp/authv1/urls.py:10]: Really sorry about that. Can you please suggest some names ?

File: BackEndApp/authv1/views.py (hunk 2 above)
Comment (mahekn23) [reply BackEndApp/authv1/views.py:109]: Will fix it.

File: BackEndApp/authv1/views.py (hunk 3 above)
Comment (mahekn23) [reply BackEndApp/authv1/views.py:140]: fixing this.

**Pull request merged**